api/views/student.py

```diff
@@ -105,6 +109,38 @@ def missed_class(self, request: Request, personal_info_id: int) -> Response:  #
         )
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @extend_schema(
+        parameters=[DashboardAvailableStudentsSerializer],
+        responses={
+            status.HTTP_200_OK: OpenApiResponse(
+                response=ListSerializer(child=MinifiedStudentSerializer()),
+                description="Students returned",
+            ),
+            status.HTTP_400_BAD_REQUEST: OpenApiResponse(
+                response=ValidationErrorSerializer,
+                description="Something is wrong with the query params",
+            ),
+            status.HTTP_404_NOT_FOUND: OpenApiResponse(
+                response=BaseAPIExceptionSerializer,
+                description="Some of time slots not found",
+            ),
```

Comment:
I guess I still don't understand what it means that "some of the time slots [were] not found" 😕 Does it mean that IDs of time slots are wrong? In this case it should be 400 bad request as well. I don't think this case is covered by tests so it's hard to tell exactly what I'm not getting :)

Reply:
How about status code [422 Unprocessable Content](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422)? Seems fitting:

> The server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions.

Reply:
I don't think this case is covered by tests BTW.

Reply:
Okay then